### PR TITLE
Vb 3698 create new session vsip endpoint that includes filters

### PR DESCRIPTION
**Pull request by dhirajshettymoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/OrchestrationSessionsController.kt [hunk 1]
```diff
@@ -77,6 +79,47 @@ class OrchestrationSessionsController(private val visitSchedulerService: VisitSc
   ): List<VisitSessionDto>? =
     visitSchedulerService.getVisitSessions(prisonCode, prisonerId, min, max)
 
+  @PreAuthorize("hasRole('ROLE_ORCHESTRATION_SERVICE__PVB')")
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/OrchestrationSessionsController.kt:82]: will wait to merge till we get consensus on Role and Rolename.

Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/OrchestrationSessionsController.kt]: updated role to VISIT_SCHEDULER. wlil update all roles to VSIP_ORCHESTRATION_SERVICE as part of a different ticket

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/OrchestrationSessionsController.kt [hunk 2]
```diff
@@ -77,6 +79,47 @@ class OrchestrationSessionsController(private val visitSchedulerService: VisitSc
   ): List<VisitSessionDto>? =
     visitSchedulerService.getVisitSessions(prisonCode, prisonerId, min, max)
 
+  @PreAuthorize("hasRole('VISIT_SCHEDULER')")
+  @GetMapping("/visit-sessions/available")
+  @Operation(
+    summary = "Returns only visit sessions for a prisoner which are within the reservable time period",
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/OrchestrationSessionsController.kt:85]: nice to use the word available..

**Review by AledWEvans — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/OrchestrationSessionsController.kt (hunk 2 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/OrchestrationSessionsController.kt]: Updated - the description is now - "Returns only available visit sessions for a specified prisoner by restriction and within the reservable time period" - same as visit-scheduler

**Pull request merged**